### PR TITLE
[MPS] Handle MPS failures of test_modules.py in common_modules.py

### PR DESCRIPTION
**Pull request by razarmehr**
- Also cleaned up `test_modules.py` from skipMPS code.
- Added `skipMPS` for unsupported or failing tests on MPS backend in common_modules.py.
   (We'll remove `skipMPS` from those tests once a fix is available for them.)

### CODE REVIEW

**Review by kulinseth — APPROVED** [review]

**Comment by kulinseth** [comment]
Overall looks good to me. @albanD can you please take a look ? This will be helpful to increase our testing coverage.

cc @malfet

File: torch/testing/_internal/common_device_type.py [hunk 1]
```diff
@@ -532,12 +531,41 @@ def setUpClass(cls):
...
+        # get a list of types provided by @dtypes or @dtypesIfMPS
+        test_dtypes = super()._get_dtypes(test)
+        if test_dtypes is not None:
+            requested_dtypes = list(test_dtypes)
+        if not requested_dtypes:
+            return None
+
+        from torch.testing._internal.common_dtype import integral_types_and
+        # populate list of datatypes supported on MPS device
+        supported_mps_dtypes = set(integral_types_and(torch.bool, torch.float16, torch.float32))
```
Comment (albanD) [inline torch/testing/_internal/common_device_type.py:562]: I would prefer that we have an explicit list of non-supported devices. As it is written here, it is non-trivial to know what is ignored.

**Review by albanD — COMMENTED** [review]
Looks great!
Could you remind me what is running on the regular m1 jobs and on the mps jobs? Is test_modules running on regular m1 test jobs and so this is covered?

File: torch/testing/_internal/common_device_type.py (hunk 1 above)
Comment (albanD) [reply torch/testing/_internal/common_device_type.py]: Ok sounds good.

**Comment by albanD** [comment]
I'm still curious whether or not this is running in CI today?

**Comment by razarmehr** [comment]
> I'm still curious whether or not this is running in CI today?

Is enabling test_modules.py for CIs something that I should do in this PR? If you or @malfet can point out where I should make that change, then I would.

**Comment by albanD** [comment]
Yes we should not land code that is not tested for sure ;) 
@kulinseth are you familiar with what currently runs? I don't know from the top of my head, we would need to check the workflows and what they run since they don't go through the general run_test (IIRC).

**Review by albanD — APPROVED** [review]
Ho nice! Thanks!
Let's see how happy CI is about this!

**Comment by razarmehr** [comment]
> Yes we should not land code that is not tested for sure ;) @kulinseth are you familiar with what currently runs? I don't know from the top of my head, we would need to check the workflows and what they run since they don't go through the general run_test (IIRC).

I believe [this commit](https://github.com/pytorch/pytorch/pull/95334/commits/a8d7987ea27d8406d0c153421dd6dd10493ddbba) should enable running the test_modules for MPS CIs. I'll monitor the CIs in this PR to confirm it.

**Comment by kulinseth** [comment]
> Ho nice! Thanks!
> Let's see how happy CI is about this!

@albanD , I think we should separate out the CI update in a separate PR. I am not sure we have enough HW yet on it. @razarmehr , can you please not combine the testing enablement in the same PR ?

**Comment by kulinseth** [comment]
@razarmehr can you take a look what's failing now ?

We have enabled more HW, and now we are ready to merge the PR.

**Comment by razarmehr** [comment]
> @razarmehr can you take a look what's failing now ?
> 
> We have enabled more HW, and now we are ready to merge the PR.

There are multiple MPS failures with the newly added tests that need to be fixed before merging this PR.

File: test/run_test.py [hunk 2]
```diff
@@ -1130,7 +1130,7 @@ def get_selected_tests(options):
         selected_tests = [tname for tname in selected_tests if tname in FUNCTORCH_TESTS]
 
     if options.mps:
-        selected_tests = ["test_mps", "test_metal"]
+        selected_tests = ['test_mps', 'test_metal', 'test_modules']
```
Comment (malfet) [inline test/run_test.py:1133]: Please use double quotes, per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings
```suggestion
        selected_tests = ["test_mps", "test_metal", "test_modules"]
```

**Pull request closed without merging**